### ISSUE TITLE
[minifier] Accuracy minification

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#1526]
- [ ] For integer tensors, currently we have torch.zeros for rand_strided (https://github.com/pytorch/torchdynamo/blob/main/torchdynamo/testing.py#L254). This is ok for Exceptions, but not really useful for accuracy. We need to get the maximum value from the original tensor and then replace zeros with randint(max_value)
- [ ] We currently do fwd_output.sum().backward() - This could hide indexing issues, because the loss is same for all the tensors. We should provide torch.rand_like() input gradients. More context here - https://github.com/pytorch/torchdynamo/pull/1414#issuecomment-1262801213

cc @williamwen42 @yanboliang

**Comment by anijain2305** [comment]
cc @msaroufim if you are looking for a task to learn about minifiers.
